examples/guide/main.go

```diff
@@ -30,9 +30,9 @@ func main() {
 		// http://<test-client>:8080/?behavior=dance
 		behavior := r.FormValue("behavior")
 
-		// when client is called with no arguments,
-		// report back with a 200 when ready to run tests
-		if behavior == "" {
+		// when method is HEAD, report back with a 200 when ready to run tests
+		if r.Method == "HEAD" {
+			w.Header().Add("Content-Length", "0")
```

Comment:
A `HEAD` request never has a body -- by definition of how HEAD is defined:
https://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html#sec9.4
```
The HEAD method is identical to GET except that the server MUST NOT return a message-body in the response. 
```

It looks like the `Content-Length` header should be the size of the body that would have been returned if this was a `GET`. In this case, since we don't know how many bytes would have been returned, not including a `Content-Length` header at all seems to be the correct response.

See related:
http://stackoverflow.com/questions/3854842/content-length-header-with-head-requests

I verified that `google.com` doesn't return a `Content-Length` for a `HEAD` request to `/`:
```
> telnet www.google.com 80
Trying 216.58.195.228...
Connected to www.google.com.
Escape character is '^]'.
HEAD / HTTP/1.1
Host: www.google.com

HTTP/1.1 200 OK
Date: Wed, 06 Jul 2016 17:28:40 GMT
Expires: -1
Cache-Control: private, max-age=0
Content-Type: text/html; charset=ISO-8859-1
P3P: CP="This is not a P3P policy! See https://www.google.com/support/accounts/answer/151657?hl=en for more info."
Server: gws
X-XSS-Protection: 1; mode=block
X-Frame-Options: SAMEORIGIN
Set-Cookie: NID=81=wiXmbp5h9Blkq0xR1gRrz5T_B49qL2I4m00E69i6e-wnmH_l9rWrC_mPGTJ_n7JfawvcmtyWHhkLG_nc1b7mn9UHIUCEizEoLBoIHzmi6Bsp5p2PWfv09RshimyY2YjXil_Iz3qF6ALiCQ; expires=Thu, 05-Jan-2017 17:28:40 GMT; path=/; domain=.google.com; HttpOnly
Transfer-Encoding: chunked
Accept-Ranges: none
Vary: Accept-Encoding
```

So I'm very suspicious that this is required for our use case. Let's discuss offline to make sure the issue isn't somewhere else.